lib/SILOptimizer/Utils/KeyPathProjector.cpp

```diff
@@ -230,7 +230,7 @@ class GettablePropertyProjector : public ComponentProjector {
       auto addr = builder.createAllocStack(loc, type);
       
       assertHasNoContext();
-      assert(getter->getArguments().size() == 2);
+      assert(!getter->isDefinition() || getter->getArguments().size() == 2);
```

Comment:
Why is the assertion loosened? Instead of `getter->getConventions().getNumSILArguments()`